File: 04.Simulation模拟/leetcode_3622_checkDivisibility.py
class Solution:
    def checkDivisibility(self, n: int) -> bool:
        s, m = 0, 1
        for i in map(int, list(str(n))): s, m = s+i, m*i
        return n%(s+m)==0

if __name__ == "__main__":
    s = Solution()
    n = 23
    print(s.checkDivisibility(n))

File: 04.Simulation模拟/leetcode_3622_checkDivisibility.cpp
#include <iostream>
using namespace std;

class Solution {
public:
    bool checkDivisibility(int n) {
        int s=0, m=1, origin=n;
        while (n){s=s+(n%10); m=m*(n%10); n=n/10;}
        return origin%(s+m)==0;
    }
};

int main(){
    Solution s;
    int n = 23;
    cout << s.checkDivisibility(n) << endl;
}
